odoo/addons/base/models/ir_actions.py

```diff
@@ -320,14 +320,23 @@ def _check_view_mode(self):
     groups_id = fields.Many2many('res.groups', 'ir_act_window_group_rel',
                                  'act_id', 'gid', string='Groups')
     search_view_id = fields.Many2one('ir.ui.view', string='Search View Ref.')
+    topbar_actions = fields.One2many('ir.actions.topbar', 'parent_action_id', string='Topbar actions linked to this action.')
     filter = fields.Boolean()
 
     def read(self, fields=None, load='_classic_read'):
         """ call the method get_empty_list_help of the model and set the window action help message
         """
         result = super(IrActionsActWindow, self).read(fields, load=load)
         if not fields or 'help' in fields:
+            topbar_actions = self.topbar_actions.filtered(
+                lambda x: (
+                    x.parent_res_id in (False, self.env.context.get('active_id', False)) and
+                    x.user_id.id in (False, self.env.uid) and
+                    x.is_visible
+                )
+            ).read()
```

Comment:
If we define a domain in the field definition then we no longer need that filtered

```suggestion
            topbar_actions = self.topbar_actions.read()
```